ex04.ex

```diff
@@ -75,7 +83,28 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd([h|t]) do
```

Comment:
## -4

There's a lot of duplication here, both in the code and in the fact that you process the list twice.

The comments made a strong suggestion that `reduce` would help:

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~